x/gov/module.go

```diff
@@ -36,14 +36,16 @@ var (
 
 // AppModuleBasic defines the basic application module used by the gov module.
 type AppModuleBasic struct {
-	cdc              codec.Codec
-	proposalHandlers []govclient.ProposalHandler // proposal handlers which live in governance cli and rest
+	cdc                    codec.Codec
+	legacyProposalHandlers []govclient.ProposalHandler // legacy proposal handlers which live in governance cli and rest
+	proposalHandlers       []govclient.ProposalHandler // proposal handlers which live in governance cli and rest
```

Comment:
`legacyProposalHandlers` CLI handlers will get added as subcommands of `NewCmdSubmitLegacyProposal` (so we can have `simd tx gov submit-legacy-proposal software-upgrade` for instance)
same for `proposalHandlers` and `NewCmdSubmitLegacyProposal` (eg for having `simd tx gov submit-proposal software-upgrade`)
see `NewTxCmd`

Reply:
so you're suggesting to have the commands available directly from `tx upgrade` command (instead of `tx gov submit-proposal`) for the new upgrade proposal msgs
but keep the legacy ones using `simd tx gov submit-legacy-proposal software-upgrade / cancel-software-upgrade`?
I kind of like this new approach, although I'm a bit concerned it might be confusing for users, any thoughts? @cmwaters @amaurym

Reply:
I'm also in favor of the new approach!

Reply:
that sounds good to me, as long as the docs/changelog are very clear about this

Reply:
Hmm on second thought this is a bit more difficult than I thought because the upgrade module doesn't really know who the authority is and whether to submit a proposal or simply execute it